students/hatims/class02/Lesson_02_Assignment.py

```diff
@@ -19,9 +19,22 @@ def __next__(self):
             if x[1] == 'Ed Sheeran':
                 return x[0]			
 				
-				
+def capture_high_energy_tracks(energy_level=0.8):
```

Comment:
how is this different from writing:
```
def capture_high_energy_tracks(energy_level=0.8):
        track =[]
        music = pd.read_csv("featuresdf.csv")
        names = [name for name in music.name]
        energy = [energy for energy in music.energy]
        myList = zip(names, energy)
        for x in myList:
            if x[1] > energy_level:
                track.append(x[0])
    return track
```

Reply:
This is a nested function where the second, function def isn't need. For it to be a closure you need to have:
```
def func1(something):
    def func2(something_else):
        var = something_else + something
        return var
   return func2
```